tests/v2/validation/clustertemplates/provisioning/provisioning_clustertemplate_test.go

```diff
@@ -0,0 +1,295 @@
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TestClusterTemplateWithQuestionsAndAnswers() {
+	log.Info("Creating an rke template with questions")
+
+	clusterTemplate, err := clustertemplates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	answers := &mgmt.Answer{Values: map[string]string{
+		"rancherKubernetesEngineConfig.kubernetesVersion":                        ct.provisioningConfig.RKE1KubernetesVersions[0],
+		"rancherKubernetesEngineConfig.network.plugin":                           ct.provisioningConfig.CNIs[0],
+		"rancherKubernetesEngineConfig.services.etcd.backupConfig.intervalHours": "10",
+	}}
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: ct.provisioningConfig.RKE1KubernetesVersions[0],
+		},
+		},
+		Questions: Questions,
+	}
+
+	clusterTemplateRevision, err := clustertemplates.CreateRkeTemplateRevision(ct.client, templateRevisonConfig, clusterTemplate.ID)
+	require.NoError(ct.T(), err)
+
+	rke1Provider := provisioning.CreateRKE1Provider(ct.provisioningConfig.Providers[0])
+	nodeTemplate, err := rke1Provider.NodeTemplateFunc(ct.client)
+	require.NoError(ct.T(), err)
+
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+
+	clusterObj, err := provisioning.CreateProvisioningRKE1ClusterWithClusterTemplate(ct.client, clusterTemplate.ID, clusterTemplateRevision.ID,
+		nodeAndRoles, nodeTemplate, answers)
+	require.NoError(ct.T(), err)
+
+	log.Info("Verifying the rke1 cluster comes up active and the cluster object has the same answer values provided.")
+
+	expectedHours := answers.Values["rancherKubernetesEngineConfig.services.etcd.backupConfig.intervalHours"]
+	actualIntervalHoursAnswers := strconv.FormatInt(clusterObj.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.IntervalHours, 10)
+	require.Equal(ct.T(), actualIntervalHoursAnswers, expectedHours)
+
+	clusterConfig := clusters.ConvertConfigToClusterConfig(ct.provisioningConfig)
+	clusterConfig.KubernetesVersion = ct.provisioningConfig.RKE1KubernetesVersions[0]
+	provisioning.VerifyRKE1Cluster(ct.T(), ct.client, clusterConfig, clusterObj)
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TestClusterTemplateEditAsAdmin() {
+	log.Info("Creating an rke template with two revisions")
+
+	clusterTemplate, err := clustertemplates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	rke1Versions, err := kubernetesversions.ListRKE1AllVersions(ct.client)
+	require.NoError(ct.T(), err)
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: rke1Versions[len(rke1Versions)-2],
```

Comment:
might be worth adding the following:
can you assert that:
* len(rke1Versions) > 2
* len(provisioningConfig.CNIs) > 1